Assets/Scripts/Gameplay/GameplayObjects/Character/NetworkAvatarGuidState.cs

```diff
@@ -34,6 +34,14 @@ public Avatar RegisteredAvatar
             }
         }
 
+        public void SetRandomAvatar()
+        {
+            if (IsServer)
+            {
+                AvatarGuid.Value = m_AvatarRegistry.GetRandomAvatar().Guid.ToNetworkGuid();
+            }
```

Comment:
should this throw if called from client? let's not have silent errors

Reply:
If I remove the IsServer check, we would get the typical NGO exception about clients writing to NetworkVariables. Would that be better?

Reply:
if we don't need to handle this ourselves, indeed